gammapy/maps/geom.py

```diff
@@ -882,6 +882,12 @@ def as_xerr(self):
             self.edges[1:] - self.center,
         )
 
+    @property
+    def iter_by_edges(self):
```

Comment:
OK Why not.

Reply:
I thought it's a very common pattern in Gammapy I can adapt the ~5 other cases in a follow up PR....